### PR TITLE
Pass list of files through stdin to avoid hitting ARG_MAX on some env…

### LINKED ISSUE #81400: //test/instrumentation:list_stable_metrics fails due to argument list
```
bazel-out/darwin-fastbuild/bin/test/instrumentation/list_stable_metrics.genrule_script.sh: line 2: ./bazel-out/host/bin/test/instrumentation/darwin_amd64_stripped/instrumentation: Argument list too long
```

/assign @serathius
/cc @logicalhan @brancz 
/sig instrumentation

### PR DESCRIPTION
**Pull request by serathius**
…ironments

/kind failing-test
/sig testing

Fixes #81400

```release-note
NONE
```

### CODE REVIEW

File: test/instrumentation/BUILD [hunk 1]
```diff
@@ -44,7 +45,7 @@ genrule(
         "//:all-srcs",
     ],
     outs = ["stable-metrics-list.yaml"],
-    cmd = "./$(locations :instrumentation) $(locations //:all-srcs) > $@",
+    cmd = "echo '$(locations //:all-srcs)' | ./$(locations :instrumentation) - > $@",
```
Comment (BenTheElder) [inline test/instrumentation/BUILD]: could we echo one to a line from the shell side instead?

Comment (serathius) [reply test/instrumentation/BUILD]: Do you mean like?
`./$(locations :instrumentation) - <(echo '$(locations //:all-srcs)') > $@`

Comment (serathius) [reply test/instrumentation/BUILD]: Sorry, I don't understand your intention. Can you give example?

Comment (BenTheElder) [reply test/instrumentation/BUILD]: Er, something like `for loc in $(locations //:all-srcs); do echo "$loc"; done | ./$(locations :instrumentation) -` ... then we don't need to lex on the go side... one input per line

Comment (serathius) [reply test/instrumentation/BUILD]: Thanks for suggestion! it's much better

**Comment by cblecker** [comment]
/lgtm
/approve

tested this on darwin, and it works

**Pull request merged**